core/src/main/java/io/grpc/transport/MessageDeframer.java

```diff
@@ -174,6 +208,84 @@ public void deframe(ReadableBuffer data, boolean endOfStream) {
   }
 
   /**
+   * Add messages to the delivery queue. This is only called by the transport thread.
```

Comment:
We should probably add similar docs to `deframe` indicating that it must be called only from the transport thread.